pkg/client/cache/listers.go

```diff
@@ -109,48 +110,56 @@ func (s *StoreToNodeLister) List() (machines api.NodeList, err error) {
 }
 
 // NodeCondition returns a storeToNodeConditionLister
-func (s *StoreToNodeLister) NodeCondition(conditionType api.NodeConditionType, conditionStatus api.ConditionStatus) storeToNodeConditionLister {
+func (s *StoreToNodeLister) NodeCondition(nodeConditionPairs ...NodeConditionPair) storeToNodeConditionLister {
 	// TODO: Move this filtering server side. Currently our selectors don't facilitate searching through a list so we
 	// have the reflector filter out the Unschedulable field and sift through node conditions in the lister.
-	return storeToNodeConditionLister{s.Store, conditionType, conditionStatus}
+	return storeToNodeConditionLister{s.Store, nodeConditionPairs}
+}
+
+type NodeConditionPair struct {
```

Comment:
Maybe just call this NodeCondition? Then if we add another field later the name doesn't need to be changed everywhere that it is being used.